presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -630,6 +639,18 @@ void PrestoServer::initializeVeloxMemory() {
     cache_ = cache::AsyncDataCache::create(allocator_.get(), std::move(ssd));
     cache::AsyncDataCache::setInstance(cache_.get());
     PRESTO_STARTUP_LOG(INFO) << cacheStr << " has been setup";
+
+    if (isCacheTtlEnabled()) {
+      cache::CacheTTLController::create(*cache_);
+      PRESTO_STARTUP_LOG(INFO) << fmt::format(
+          "Cache TTL is enabled, with TTL {} enforced every {}.",
+          succinctMillis(duration_cast<milliseconds>(
+                             systemConfig->cacheVeloxTtlThreshold())
+                             .count()),
+          succinctMillis(duration_cast<milliseconds>(
+                             systemConfig->cacheVeloxTtlCheckInterval())
+                             .count()));
```

Comment:
Yeah do sanity check inside here. And let the server crash by throwing if check fails

Reply:
Revised